Source/Core/GroundPolylineGeometry.js

```diff
@@ -0,0 +1,1054 @@
+        var coplanarPlane = Plane.fromPointNormal(vertexBottom, coplanarNormal, coplanarPlaneScratch);
+        var nextBottomDistance = Plane.getPointDistance(coplanarPlane, nextBottom);
+        if (CesiumMath.equalsEpsilon(nextBottomDistance, 0.0, CesiumMath.EPSILON7)) {
+            // If the points are coplanar, point the normal in the direction of the plane
+            Cartesian3.clone(coplanarNormal, result);
+            return result;
+        }
+
+        // Average directions to previous and to next
+        result = Cartesian3.add(toNext, toPrevious, result);
+        result = Cartesian3.multiplyByScalar(result, 0.5, result);
+        result = Cartesian3.normalize(result, result);
+
+        // Rotate this direction to be orthogonal to up
+        var forward = Cartesian3.cross(up, result, forwardScratch);
+        Cartesian3.normalize(forward, forward);
+        Cartesian3.cross(forward, up, result);
+        Cartesian3.normalize(result, result);
+
+        // Flip the normal if it isn't pointing roughly bound right (aka if forward is pointing more "backwards")
+        if (Cartesian3.dot(toNext, forward) < cosine90) {
+            result = Cartesian3.multiplyByScalar(result, -1.0, result);
+        }
+
+        return result;
+    }
+
+    var XZ_PLANE = Plane.fromPointNormal(Cartesian3.ZERO, Cartesian3.UNIT_Y);
+
+    var previousBottomScratch = new Cartesian3();
+    var vertexBottomScratch = new Cartesian3();
+    var vertexTopScratch = new Cartesian3();
+    var nextBottomScratch = new Cartesian3();
+    var vertexNormalScratch = new Cartesian3();
+    var intersectionScratch = new Cartesian3();
+    /**
+     * Computes shadow volumes for the ground polyline, consisting of its vertices, indices, and a bounding sphere.
+     * Vertices are "fat," packing all the data needed in each volume to describe a line on terrain.
+     *
+     * @param {GroundPolylineGeometry} groundPolylineGeometry
+     * @private
+     */
+    GroundPolylineGeometry.createGeometry = function(groundPolylineGeometry) {
+        var loop = groundPolylineGeometry.loop;
+        var ellipsoid = groundPolylineGeometry.ellipsoid;
+        var granularity = groundPolylineGeometry.granularity;
+        var projection = new PROJECTIONS[groundPolylineGeometry._projectionIndex](ellipsoid);
+
+        var minHeight = WALL_INITIAL_MIN_HEIGHT;
+        var maxHeight = WALL_INITIAL_MAX_HEIGHT;
+
+        var index;
+        var i;
+
+        var positions = groundPolylineGeometry.positions;
+        var positionsLength = positions.length;
+
+        //>>includeStart('debug', pragmas.debug);
+        if (positionsLength < 2) {
+            throw new DeveloperError('GroundPolylineGeometry must contain two or more positions');
+        }
+        //>>includeEnd('debug');
+        if (positionsLength === 2) {
+            loop = false;
+        }
+
+        // Split positions across the IDL and the Prime Meridian as well.
```

Comment:
The vertex attributes include the start and end positions of the line in 3D and in 2D, so the geometry has to be pre-split for these to get computed correctly.
I had some really funny early debug stuff where a a segment split across the IDL had the correct shadow volume but the vertex attributes thought the line segment should go from -179.99 to 179.99 over the greater arc...

The split along the prime meridian as well was because `GeometryPipeline` splits along a plane that covers both the IDL and the Prime meridian, and it's possible to have a bounding sphere that doesn't actually cross the IDL but does trigger the split code.

Even thought the Prime meridian wouldn't have the same 2D problem we'd still have overdraw in some views, because there would be two adjacent volumes that both encode the same line segment.
Figured it'd just be easier to stay away from the Prime meridian and the IDL altogether.